File: RSA Private Key Finder.cpp
// Coding with Mubeen Nawaz.
// C++ Program to find Private Key in RSA.
// When n < 1000000.
#include <iostream>
using namespace std;
int main(){
	float n, e, p, q, a, d, d2;
	int	d1, i = 0;
	cout<<"Enter p: ";
	cin>>p;
	cout<<"Enter q: ";
	cin>>q;
	cout<<"Enter Public Key: ";
	cin>>e;
	n = p * q;
	a = (p-1)*(q-1);
	cout<<"\n*************************\n";
	do{
		d = (1 + a * i) / e;
		cout<<"\n"<<d<<"\n";
		d1 = d;
		d2 = d - d1;
		i++;
	}while(d2 != 0);
	cout<<"\n*************************\n"
	    <<"\nn is: "<<n<<"\n"
		<<"\nPhy(n) is: "<<a<<"\n"
	    <<"\nPrivate Key is: "<<d<<"\n";
	return 0;
}

File: RSA Private Key Finder.py
# Coding with Mubeen Nawaz
# RSA Private Key Finder
p = float(input("Enter value of p: "))
q = float(input("Enter value of q: "))
e = float(input("Enter value of e: "))
print("*****************************")
def gcd(a, b):
    if a == 0:
        return (b, 0, 1)
    else:
        g, y, x = gcd(b % a, a)
        return (g, x - (b // a) * y, y)
def inverse(a, m):
    g, x, y = gcd(a, m)
    if g != 1:
        print("\nInverse not exist\n")
        exit(0)
    else:
        return x % m
def private_key(p, q, e):
    n = p * q
    phi_n = (p - 1) * (q - 1)
    d = inverse(e, phi_n)
    return (n, phi_n, d)
n, phi_n, d = private_key(p, q, e)
a = ((d * e) - 1) / phi_n
print("\nValue of n:", n)
print("\nValue of Phi(n):", phi_n)
print("\nValue of a: ", a)
print("\nPrivate key (d):", d, "\n")